app/frontend/src/features/messages/groupchats/groupChatsSlice.ts

```diff
@@ -0,0 +1,88 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+import {
+  fetchGroupChatsThunk,
+  fetchMessagesThunk,
+  leaveGroupChatThunk,
+  sendMessageThunk,
+} from ".";
+import { GroupChat, Message } from "../../../pb/conversations_pb";
+
+const initialState = {
+  groupChats: [] as GroupChat.AsObject[],
+  error: "",
+  loading: false,
+  groupChatView: {
+    groupChat: null as GroupChat.AsObject | null,
+    messages: [] as Message.AsObject[],
+    error: "",
+    loading: false,
+  },
+};
+
+export type RequestsState = typeof initialState;
+
+export const groupChatsSlice = createSlice({
+  name: "groupChats",
+  initialState,
+  reducers: {
+    groupChatsFetched(state, action: PayloadAction<GroupChat.AsObject[]>) {
+      state.groupChats = action.payload;
+    },
+    setGroupChat(state, action: PayloadAction<GroupChat.AsObject | null>) {
+      state.groupChatView.groupChat = action.payload;
+    },
+    setMessages(state, action: PayloadAction<Message.AsObject[]>) {
+      state.groupChatView.messages = action.payload;
+    },
+  },
+  extraReducers: (builder) => {
+    builder
+      .addCase(fetchGroupChatsThunk.pending, (state) => {
+        state.error = "";
+        state.loading = true;
+      })
+      .addCase(fetchGroupChatsThunk.fulfilled, (state) => {
+        state.loading = false;
+      })
+      .addCase(fetchGroupChatsThunk.rejected, (state, action) => {
+        state.error = action.error.message || "";
```

Comment:
I think `action.error.message!` is better so there would be an error even though a null message shouldn't be possible.

Reply:
What error? From Typescript? Do you mean instead of ` || ""` ?

Reply:
state.error is a string. Should I change that to `string || null` ?

Reply:
```suggestion
        state.error = action.error.message!;
```
Like this. I assumed `|| ""` was to satisfy the typing, but if somehow the thunk is rejected but `action.error.message` is null, that would silently fail with no visible error.

Reply:
That cannot be null: https://tc39.es/ecma262/#sec-error.prototype.message if it is a normal JS error.

Reply:
If it's rejected, something went wrong. If something went wrong, there should either be an error message displayed or an exception raised. As it is, if error.message is null, it looks to the user like fetchGroupChats should have worked but nothing happens.

Maybe it doesn't matter though if message always exists (I think it does for gRPC errors too)